### DISCUSSION TITLE
Remap to Gyro controls?

### DISCUSSION THREAD

**Discussion by BrokenPhysics13** [hax0kartik/rehid#36] [Ideas]
I was wondering, would be possible to add remapping of the touch screen to gyro? A game in particular this I think would benefit from this is Kid Icarus: Uprising, which normally uses the touchscreen (or face buttons) to aim. With rehid in its current form you can remap the touch controls to the c-stick for something close to have dual-analog, which works so-so, but is not perfect and is nigh impossible on the hardest difficulties without using touch controls. However, I think if we could map the touch screen to the gyroscope for aiming, the game would be much easier to play! I'm aware of a project that uses Twilight Menu++ and action replay codes to alter controls for DS games like Metroid Prime Hunters (which also uses the touch screen to aim), and it adds use of circle pad for analog AND gyro controls for aiming on N3DS systems (https://gbatemp.net/threads/circle-pad-patches-for-super-mario-64-ds-and-other-games-in-twilightmenu-with-twpatcher-and-rtcom.623267/)! Again, would there be a way to do something like this for rehid?

**Comment by hax0kartik** [comment]
Hi, while this is not impossible to do with rehid, I find the maths around gyro really complex..(quaternions and other stuff). I am, ofcourse, happy to help anyone make changes to rehid code who knows how to deal with the maths...

**Comment by MetaKnightAresX** [comment]
Yes I also wish the c-stick could be mapped to where when it moved the touch screen would count it to. I would like to say that it is hard to play Kid Icarus: Uprising without it. I am looking for a way so that ALL of the touch screen gets moved when the c-stick moves too. I use https://mikahjc.github.io/3dsRemapBuilder/config  for these and I find it that it only moves the c-stick to where we set it and not move around, so if there is a way to do this pleas respond, I really need this 😭
